controllers/volumereplicationgroup_controller.go

```diff
@@ -519,8 +600,13 @@ func (v *VRGInstance) processAsPrimary() (ctrl.Result, error) {
 		return ctrl.Result{Requeue: true}, nil
 	}
 
-	requeue := false
-	requeue = v.reconcileVRsAsPrimary()
+	if err := v.restorePVs(); err != nil {
+		v.log.Error(err, "Restoring PVs failed")
+		// Since updating status failed, reconcile
+		return ctrl.Result{Requeue: true}, nil
```

Comment:
Do we want to indicate this (i.e. restoring PV failed) in VRG status? IIUC we cannot change the status.conditions.Available to false as the restorePVs mainly depends on VRG being available. However, I am just wondering if capturing this information in the status (where status.conditions[available] is true, but message says that restoring PVs failed.) 

Even without the above point, we can still get that information by looking at VRG controller logs. Just wondering if capturing it in VRG status would be any more helpful.

Let me know your thoughts on this.

Reply:
This can be useful, to have a separate condition, type: MetadataAvailable; reason: restored|progressing|error; status: true(restored)|false(progressing|error)